File: backend/src/main/java/com/sprachwelt/model/Word.java
package com.sprachwelt.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "word")
public class Word {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "text_id", nullable = false)
    private Text text;

    private String content;
    private Integer position;

    private WordStatus status;

    private WordType type;
}

File: frontend/src/app/model/word.ts
import { WordStatus } from './word-status';

export class Word {
  id: string;
  text: string;
  position?: number;
  status?: WordStatus;

  constructor(id: string, text: string, position: number, status: WordStatus) {
    this.id = id;
    this.text = text;
    this.position = position;
    this.status = status;
  }
}
